libavpipe/src/avpipe_xc.c

```diff
@@ -4921,13 +4927,31 @@ avpipe_fini(
         }
     }
 
+    if ((*xctx)->params->copy_mpegts) {
+        cp_ctx_t *cp_ctx = &(*xctx)->cp_ctx;
+        coderctx_t *mpegts_encoder_ctx = &cp_ctx->encoder_ctx;
+        elv_warn("Freeing mpegts encoder context, closing io");
+        avio_close(mpegts_encoder_ctx->format_context->pb);
+        avformat_free_context(mpegts_encoder_ctx->format_context);
```

Comment:
Is there no `avpipe_opaque` to free in this case?

Reply:
There is, freed and comment added.